esphome/components/esp32/__init__.py

```diff
@@ -670,6 +675,9 @@ def _validate_idf_component(config: ConfigType) -> ConfigType:
                         CONF_ENABLE_LWIP_CHECK_THREAD_SAFETY, default=True
                     ): cv.boolean,
                     cv.Optional(CONF_EXECUTE_FROM_PSRAM): cv.boolean,
+                    cv.Optional(CONF_MAIN_LOOP_STACK_SIZE, default=8192): cv.int_range(
+                        MIN_MAIN_LOOP_TASK_STACK_SIZE, 1024**6
```

Comment:
```suggestion
                        MIN_MAIN_LOOP_TASK_STACK_SIZE, MAX_MAIN_LOOP_TASK_STACK_SIZE
```

1152921504606846976 might be a bit much

Reply:
Added a fixed upper size to 128KB. As this is allocated from the 160KB heap it should not be increased indefinitely.

Eventually to investigate there should be some checkmarks added https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/performance/ram-usage.html#run-time-methods-to-determine-stack-size

I will think about adding some sort of watchdog task to report when the marks are filling. At the moment the generic stack trace gives no hint to whoever is filling up the stack or improve on it:
```
[08:48:25]***ERROR*** A stack overflow in task loopTask has been detected.


[08:48:25]Backtrace: 0x40083759:0x3ffbd950 0x4008a945:0x3ffbd970 0x4008b812:0x3ffbd990 0x4008c533:0x3ffbda10 0x4008b958:0x3ffbda30 0x4008b90a:0x000000ff |<-CORRUPTED
WARNING Found stack trace! Trying to decode it
WARNING Decoded 0x40083759: panic_abort at /Users/USER/.platformio/packages/framework-espidf/components/esp_system/panic.c:469
WARNING Decoded 0x4008a945: esp_system_abort at /Users/USER/.platformio/packages/framework-espidf/components/esp_system/port/esp_system_chip.c:87
WARNING Decoded 0x4008b812: vApplicationStackOverflowHook at /Users/USER/.platformio/packages/framework-espidf/components/freertos/FreeRTOS-Kernel/portable/xtensa/port.c:563
WARNING Decoded 0x4008c533: vTaskSwitchContext at /Users/USER/.platformio/packages/framework-espidf/components/freertos/FreeRTOS-Kernel/tasks.c:3689 (discriminator 7)
WARNING Decoded 0x4008b958: _frxt_dispatch at /Users/USER/.platformio/packages/framework-espidf/components/freertos/FreeRTOS-Kernel/portable/xtensa/portasm.S:451
WARNING Decoded 0x4008b90a: _frxt_int_exit at /Users/USER/.platformio/packages/framework-espidf/components/freertos/FreeRTOS-Kernel/portable/xtensa/portasm.S:246
```